crates/sui-sdk-types/src/types/transaction/unresolved.rs

```diff
@@ -321,3 +356,21 @@ impl From<Value> for serde_json::Value {
         }
     }
 }
+
+#[cfg(all(feature = "serde", feature = "hash"))]
+impl From<&Object> for Input {
+    fn from(object: &Object) -> Self {
+        match object.owner() {
+            Owner::Address(_) => object.as_input(),
+            Owner::Object(_) => object.as_input().with_owned_kind(),
+            Owner::Shared(at_version) => object.as_input().with_initial_shared_version(*at_version),
+            Owner::Immutable => object.as_input().with_immutable_kind(),
+        }
+    }
+}
```

Comment:
I'm having a really hard time understanding why we'd have a From impl here that doesn't match the as_input impl above. This seems very confusing and error prone. Lets get rid of one of the two of these (maybe the as_input method).

Same comment as above, inline the imports here either by fully qualifying the paths or by just having the use statements inside the function.